src/data/bucket.ts

```
@@ -114,8 +114,8 @@ export function deserialize(input: Array<Bucket>, style: Style): {[_: string]: B
         // look up StyleLayer objects from layer ids (since we don't
         // want to waste time serializing/copying them from the worker)
         (bucket as any).layers = layers;
-        if ((bucket as any).stateDependentLayerIds) {
-            (bucket as any).stateDependentLayers = (bucket as any).stateDependentLayerIds.map((lId) => layers.filter((l) => l.id === lId)[0]);
+        if (bucket.stateDependentLayerIds) {
+            (bucket as any).stateDependentLayers = bucket.stateDependentLayerIds.map((lId) => layers.filter((l) => l.id === lId)[0]);
```

Why can you remove `(bucket as any)` in one place but not in the other?

---

as mentioned in the original thread, the reason we need as any is because bucket layers are readonly. This only applies to writing to the object, reading is ok.. I assume that removing readonly from the type has further consequences that we may not want, so readonly only needs to be overridden in this specific place. There might be a better way to ignore readonly (maybe just `bucket!` to force the compiler to ignore the error). I'll try as a replacement to as any in the writable ones